File: 10.2D/Example code/Shape/Shape.java
public abstract class Shape {
    // private field
    private float _x;
    private float _y;

    // constructor
    public Shape(float x, float y) 
    {
        _x = x;
        _y = y;
    }
    // property field

    public float get_y() {
        return _y;
    }

    public void set_y(float _y) {
        this._y = _y;
    }

    public float get_x() {
        return _x;
    }

    public void set_x(float _x) {
        this._x = _x;
    }
    
    // methods
    public abstract void Draw(float x, float y);
}

File: 6.3D/6.3D/GameTemplate/src/Shape.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SwinGameSDK;

namespace TrafficController
{
    public abstract class Shape
    {
        //Private field
        private Color _color;
        private float _x;
        private float _y;
        //Property
        public Color Color { get => _color; set => _color = value; }
        public float X { get => _x; set => _x = value; }
        public float Y { get => _y; set => _y = value; }

        //Contructor
        public Shape(Color color, float x, float y)
        {
            _color = color;
            X = x;
            Y = y;
        }
        //Methods
        public abstract void Draw();
        public abstract bool MouseIn(Point2D pt);
    }
}
